deflect/NetworkListener.h

```diff
@@ -0,0 +1,92 @@
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer in the documentation and/or other materials       */
+/*      provided with the distribution.                              */
+/*                                                                   */
+/*    THIS  SOFTWARE IS PROVIDED  BY THE  UNIVERSITY OF  TEXAS AT    */
+/*    AUSTIN  ``AS IS''  AND ANY  EXPRESS OR  IMPLIED WARRANTIES,    */
+/*    INCLUDING, BUT  NOT LIMITED  TO, THE IMPLIED  WARRANTIES OF    */
+/*    MERCHANTABILITY  AND FITNESS FOR  A PARTICULAR  PURPOSE ARE    */
+/*    DISCLAIMED.  IN  NO EVENT SHALL THE UNIVERSITY  OF TEXAS AT    */
+/*    AUSTIN OR CONTRIBUTORS BE  LIABLE FOR ANY DIRECT, INDIRECT,    */
+/*    INCIDENTAL,  SPECIAL, EXEMPLARY,  OR  CONSEQUENTIAL DAMAGES    */
+/*    (INCLUDING, BUT  NOT LIMITED TO,  PROCUREMENT OF SUBSTITUTE    */
+/*    GOODS  OR  SERVICES; LOSS  OF  USE,  DATA,  OR PROFITS;  OR    */
+/*    BUSINESS INTERRUPTION) HOWEVER CAUSED  AND ON ANY THEORY OF    */
+/*    LIABILITY, WHETHER  IN CONTRACT, STRICT  LIABILITY, OR TORT    */
+/*    (INCLUDING NEGLIGENCE OR OTHERWISE)  ARISING IN ANY WAY OUT    */
+/*    OF  THE  USE OF  THIS  SOFTWARE,  EVEN  IF ADVISED  OF  THE    */
+/*    POSSIBILITY OF SUCH DAMAGE.                                    */
+/*                                                                   */
+/* The views and conclusions contained in the software and           */
+/* documentation are those of the authors and should not be          */
+/* interpreted as representing official policies, either expressed   */
+/* or implied, of The University of Texas at Austin.                 */
+/*********************************************************************/
+
+#ifndef DEFLECT_NETWORK_LISTENER_H
+#define DEFLECT_NETWORK_LISTENER_H
+
+#include <deflect/types.h>
+#include <QtNetwork/QTcpServer>
+
+namespace deflect
+{
+
+/**
+ * Listen to incoming PixelStream connections from Stream clients.
+ */
+class NetworkListener : public QTcpServer
+{
+    Q_OBJECT
+
+public:
+    /** The default port number used for Stream connections. */
+    static const int defaultPortNumber_;
+
+    /**
+     * Create a new server listening for Stream connections.
+     * @param port The port to listen on. Must be available.
+     * @throw std::runtime_error if the server could not be started.
+     */
+    explicit NetworkListener(int port = defaultPortNumber_);
+
+    /** Destructor */
+    ~NetworkListener();
+
+    /** Get the command handler. */
+    CommandHandler& getCommandHandler();
+
+    /** Get the PixelStreamDispatcher. */
+    PixelStreamDispatcher& getPixelStreamDispatcher();
+
+protected:
+    /** Re-implemented handling of connections from QTCPSocket. */
+    void incomingConnection(int socketHandle) override;
+
+signals:
+    void registerToEvents(QString uri, bool exclusive,
+                          deflect::EventReceiver* receiver);
```

Comment:
Hmmm, the API looks confusing now because the following two functions should in fact be slots. I'm wondering how we can document that? I would have added two public slots which then emit these "private" signals, but Qt don't allow private modifiers on signals, so this is still imperfect.